openedx/adg/admin_dashboard/applications/tests/test_views.py

```diff
@@ -61,3 +65,37 @@ def test_application_dashboard_view_get_request_for_admin(is_superuser, group_na
     mock_render.assert_called_once_with(
         request, 'adg/admin_dashboard/applications/applications_summary.html', context=expected_context
     )
+
+
+@pytest.mark.parametrize('page_url, app_type', [
+    ('open_applications', UserApplication.OPEN),
+    ('waitlisted_applications', UserApplication.WAITLIST),
+    ('accepted_applications', UserApplication.ACCEPTED)
+])
+@pytest.mark.django_db
+@skip_unless_lms
+def test_application_listing_view_context(
+    page_url, app_type, adg_admin, open_applications, waitlisted_applications, accepted_applications
+):
+    """
+    Test that the applicationListingView returns the correct context
+    """
+    kwargs = {'application_type': app_type}
+    listing_view_request = RequestFactory().get(reverse(page_url, kwargs=kwargs))
+    listing_view_request.user = adg_admin
+    response = ApplicationListingView.as_view()(listing_view_request, **kwargs)
+
+    received_context = response.context_data
+
+    expected_total_applications = 0
+    if app_type == UserApplication.OPEN:
+        expected_total_applications = len(open_applications)
+    elif app_type == UserApplication.WAITLIST:
+        expected_total_applications = len(waitlisted_applications)
+    else:
+        expected_total_applications = len(accepted_applications)
```

Comment:
this is not testing anything, as all the fixtures of open, waitlisted and accepted applications have length 5, and the expected_total_applications would always be 5

Reply:
Actually, applications queryset is being tested in the helpers already. I'll still change the number of each applications type just so that it makes more sense